### PR TITLE
Support aiven

### PR DESCRIPTION
**Pull request by tommpy**

### CODE REVIEW

File: src/servicetest/resources/logback.-test.xml [hunk 1]
```diff
@@ -0,0 +1,15 @@
...
+<configuration>
+
+    <appender name="CONSOLE" class="ch.qos.logback.core.ConsoleAppender">
+        <target>System.out</target>
+        <encoder>
+            <pattern>%date{ISO8601} %-5level[%thread] %logger{0} %X{} - %msg%n</pattern>
+        </encoder>
+    </appender>
+
+    <root level="TRACE">
```
Comment (laurence-bird) [inline src/servicetest/resources/logback.-test.xml]: Perhaps best to change this back to info?

File: cloudformation.yaml [hunk 2]
```diff
@@ -23,6 +23,17 @@ Resources:
...
+          PolicyName: read-aiven-bucket
+          PolicyDocument:
+            Statement:
+            - Effect: Allow
+              Action:
+                - 's3:Get*'
+                - 's3:List*'
+              Resource:
+                - 'arn:aws:s3:::ovo-comms-aiven'
+                - 'arn:aws:s3:::ovo-comms-aiven/*'
```
Comment (cb372) [inline cloudformation.yaml]: Don't forget to apply this change to both stacks before deploying

File: project/Dependencies.scala [hunk 3]
```diff
@@ -5,26 +5,27 @@ object Dependencies {
...
     "io.circe"                   %% "circe-parser"              % circeVersion,
     "io.circe"                   %% "circe-generic"             % circeVersion,
     "me.moocar"                  % "logback-gelf"               % "0.2",
+    "org.apache.kafka"           %% "kafka"                     % "0.10.2.1",
     "net.cakesolutions"          %% "scala-kafka-client"        % "0.10.0.0",
     "io.logz.logback"            % "logzio-logback-appender"    % "1.0.11",
-    "com.ovoenergy"              %% "comms-kafka-messages"      % "1.20",
-    "com.ovoenergy"              %% "comms-kafka-serialisation" % "2.3",
+    "com.ovoenergy"              %% "comms-kafka-messages"      % "1.25",
+    "com.ovoenergy"              %% "comms-kafka-serialisation" % "2.7",
```
Comment (cb372) [inline project/Dependencies.scala]: New version of comms-kafka-serialisation on the way soon... probably worth bumping to that before merging.

File: src/main/resources/prd.conf [hunk 4]
```diff
@@ -6,7 +6,28 @@ mailgun {
...
 kafka {
-  hosts = "@@{prd.kafka_hosts}" # comma-separated, e.g. "host1:9092,host2:9092"
+  schema.registry {
+    url = "@@{prd.aiven.schema_registry.url}"
+    username = "@@{prd.aiven.schema_registry.username}"
+    password = "@@{prd.aiven.schema_registry.password}"
+  }
+
+  legacy{
+    hosts = "@@{prd.mailgun.api_key}"
```
Comment (cb372) [inline src/main/resources/prd.conf]: that doesn't look right...

File: src/main/scala/com/ovoenergy/delivery/service/Main.scala [hunk 5]
```diff
@@ -149,8 +217,11 @@ object Main extends App with LoggingWithMDC {
   }
 
   log.info("Delivery Service started")
-  setupGraph(emailGraph, "Email")
-  setupGraph(smsGraph, "SMS")
+  setupGraph(emailGraphAiven, "Email")
+  setupGraph(smsGraphAiven, "SMS")
+
+  setupGraph(emailGraphLegacy, "Email")
+  setupGraph(smsGraphLegacy, "SMS")
```
Comment (cb372) [inline src/main/scala/com/ovoenergy/delivery/service/Main.scala]: Might be useful to include the word "Aiven"/"legacy" in the description that gets included in the error log when it dies

File: src/main/scala/com/ovoenergy/delivery/service/kafka/DeliveryServiceGraph.scala [hunk 6]
```diff
@@ -49,11 +53,23 @@ object DeliveryServiceGraph extends LoggingWithMDC {
         Supervision.Stop
     }
 
-    val consumerSettings =
+    val initialSettings =
       ConsumerSettings(actorSystem, new StringDeserializer, consumerDeserializer)
         .withBootstrapServers(kafkaConfig.hosts)
         .withGroupId(kafkaConfig.groupId)
 
+    val consumerSettings = sslConfig.fold(initialSettings) { ssl =>
```
Comment (cb372) [inline src/main/scala/com/ovoenergy/delivery/service/kafka/DeliveryServiceGraph.scala]: Could use the `comms-kafka-akka-helpers` lib to do this

File: src/main/scala/com/ovoenergy/delivery/service/kafka/Publisher.scala [hunk 7]
```diff
@@ -2,19 +2,45 @@ package com.ovoenergy.delivery.service.kafka
...
 object Publisher {
+  import com.ovoenergy.kafka.serialization.avro4s._
+
+  def producerFor[T: SchemaFor: ToRecord](kafkaConfig: KafkaConfig, sslConfigMaybe: Option[SSLConfig])(
+      implicit schemaRegistrySettings: SchemaRegistryClientSettings) = {
+    val initialSettings = Conf(new StringSerializer,
+                               avroBinarySchemaIdSerializer[T](schemaRegistrySettings, isKey = false),
+                               kafkaConfig.hosts)
+
+    val producerSettings = sslConfigMaybe
```
Comment (cb372) [inline src/main/scala/com/ovoenergy/delivery/service/kafka/Publisher.scala]: Similarly here, could use the `comms-kafka-cakesolutions-helpers` lib

File: src/servicetest/resources/logback.-test.xml [hunk 8]
```diff
@@ -0,0 +1,15 @@
+<?xml version="1.0" encoding="UTF-8"?>
+<configuration>
+
```
Comment (cb372) [inline src/servicetest/resources/logback.-test.xml]: What's up with the weird name for this file?

**Comment by cb372** [comment]
You can remove `sbt-docker-compose` from plugins.sbt, and you might need to update the readme accordingly

File: src/servicetest/resources/logback.-test.xml (hunk 8 above)
Comment (tommpy) [inline src/servicetest/resources/logback.-test.xml]: Sorted

File: src/main/scala/com/ovoenergy/delivery/service/kafka/Publisher.scala (hunk 7 above)
Comment (tommpy) [inline src/main/scala/com/ovoenergy/delivery/service/kafka/Publisher.scala]: Postponing until the legacy gets ripped out

File: src/main/scala/com/ovoenergy/delivery/service/kafka/DeliveryServiceGraph.scala (hunk 6 above)
Comment (tommpy) [inline src/main/scala/com/ovoenergy/delivery/service/kafka/DeliveryServiceGraph.scala]: Postponing until the legacy gets ripped out

File: src/main/scala/com/ovoenergy/delivery/service/Main.scala (hunk 5 above)
Comment (tommpy) [inline src/main/scala/com/ovoenergy/delivery/service/Main.scala]: Done

File: src/main/resources/prd.conf (hunk 4 above)
Comment (tommpy) [inline src/main/resources/prd.conf]: Sorted

File: project/Dependencies.scala (hunk 3 above)
Comment (tommpy) [inline project/Dependencies.scala]: Done

File: src/servicetest/resources/logback.-test.xml (hunk 1 above)
Comment (tommpy) [inline src/servicetest/resources/logback.-test.xml]: Bosched

**Pull request merged**